lib/bootstrap_form/components/validation.rb

```diff
@@ -8,21 +8,23 @@ module Validation
       private
 
       def error?(name)
-        object.respond_to?(:errors) && !(name.nil? || object.errors[name].empty?)
+        name && object.respond_to?(:errors) &&
+          (object.errors[name].any? || (name.end_with?("_id") && object.errors[name[0..-4]].any?))
       end
 
       def required_attribute?(obj, attribute)
         return false unless obj && attribute
 
         target = obj.instance_of?(Class) ? obj : obj.class
+        return false unless target.respond_to? :validators_on
 
-        target_validators = if target.respond_to? :validators_on
-                              target.validators_on(attribute).map(&:class)
-                            else
-                              []
-                            end
+        presence_validator?(target_validators(target, attribute))
+      end
 
-        presence_validator?(target_validators)
+      def target_validators(target, attribute)
+        target_validators = target.validators_on(attribute).map(&:class)
+        target_validators.concat target.validators_on(attribute[0..-4]).map(&:class) if attribute.end_with?("_id")
```

Comment:
I believe an association can specify any string for the foreign key, so I wonder if just looking for "_id" at the end of the attribute is going to cover all cases. I'm guessing there's a way to access the foreign key column name for an association.

Reply:
Absolutely, but I wanted to cover the most common case first to get the big win for little work 😄 

Can we put it in a separate issue/PR?

Reply:
Good plan. Please create the issue and reference it here. I'll approve and merge. Are you going to work on the more generalized solution?